api/types/resource_ids.go

```diff
@@ -80,20 +80,16 @@ func ResourceIDToString(id ResourceID) string {
 // ResourceIDFromString parses a ResourceID from a string. The string should
 // have been obtained from ResourceIDToString.
 func ResourceIDFromString(raw string) (ResourceID, error) {
-	if len(raw) < 1 || raw[0] != '/' {
-		return ResourceID{}, trace.BadParameter("%s is not a valid ResourceID string", raw)
-	}
-	raw = raw[1:]
 	// Should be safe for any Name as long as the ClusterName and Kind don't
 	// contain slashes, which should never happen.
-	parts := strings.SplitN(raw, "/", 3)
-	if len(parts) != 3 {
-		return ResourceID{}, trace.BadParameter("/%s is not a valid ResourceID string", raw)
+	parts := strings.SplitN(raw, "/", 4)
```

Comment:
Why 4?

Reply:
The `strings.SprintN` makes a blank initial entry before the first `/` so it is 4 parts.

Reply:
Ohh, I see you removed the part that trims the leading `/`.

Its a bit odd that you are operating on `raw` first, and then later doing some validity checks on raw after you've already split it.

I'd find it cleaner to do input validation prior to operating on the input.

Reply:
That's fair. Nice to not repeat the same message.